### PR TITLE
Update config options

### PR DESCRIPTION
**Pull request by jiasli**
Update `core.output` config option to align with the in-tool help:

```
> az account show -h
...
    --output -o                 : Output format.  Allowed values: json, jsonc, none, table, tsv,
                                  yaml, yamlc.  Default: none.
```

Wording for other options is also unified.

### CODE REVIEW

File: docs-ref-conceptual/azure-cli-configuration.md [hunk 1]
```diff
@@ -98,7 +98,7 @@ When you provide a default value, that argument is no longer required by any com
 
 | Section | Name      | Type | Description|
 |---------|-----------|------|------------|
-| __core__ | output | string | The default output format. Can be one of `json`, `jsonc`, `tsv`, or `table`. |
+| __core__ | output | string | The default output format. Allowed values: `json` (default), `jsonc`, `tsv`, `yaml`, `yamlc`, `table`, `none`. |
```
Comment (jiasli) [inline docs-ref-conceptual/azure-cli-configuration.md:101]: The in-tool help shows allowed values alphabetically:

> Allowed values: json, jsonc, none, table, tsv, yaml, yamlc.

I changed the order in the document based on their use frequency.

**Review by dbradish-microsoft — APPROVED** [review]

**Review by dcaro — APPROVED** [review]
LGTM

**Review by yonzhan — APPROVED** [review]

File: docs-ref-conceptual/azure-cli-configuration.md [hunk 2]
```diff
@@ -98,7 +98,7 @@ When you provide a default value, that argument is no longer required by any com
 
 | Section | Name      | Type | Description|
 |---------|-----------|------|------------|
-| __core__ | output | string | The default output format. Can be one of `json`, `jsonc`, `tsv`, or `table`. |
+| __core__ | output | string | The default output format. Allowed values: `json` (default), `jsonc`, `yaml`, `yamlc`, `table`, `tsv`, `none`. |
```
Comment (jiasli) [inline docs-ref-conceptual/azure-cli-configuration.md:101]: I resorted the allowed values to align with https://learn.microsoft.com/en-us/cli/azure/format-output-azure-cli. I think we can also add:

```suggestion
| __core__ | output | string | The default output format. Allowed values: `json` (default), `jsonc`, `yaml`, `yamlc`, `table`, `tsv`, `none`. For more information, see [Output formats for Azure CLI commands](format-output-azure-cli.md). |
```

File: .docsreference.azurecli.json [hunk 3]
```diff
@@ -93,7 +93,7 @@
             "isRequired": false,
             "summary": "Output format.",
             "defaultValue": "json",
-            "parameterValueGroup": "json, jsonc, table, tsv"
+            "parameterValueGroup": "json, jsonc, none, table, tsv, yaml, yamlc"
```
Comment (jiasli) [inline .docsreference.azurecli.json:96]: I use alphabetic order to align with the in-tool help:

```
> az account show -h
...
    --output -o                 : Output format.  Allowed values: json, jsonc, none, table, tsv,
                                  yaml, yamlc.  Default: none.
```

Comment (jiasli) [inline .docsreference.azurecli.json:96]: Also, should we link to `format-output-azure-cli.md` in the reference doc here?

**Review by jsntcy — APPROVED** [review]

**Review by yonzhan — APPROVED** [review]

**Pull request merged**